### PR TITLE
consolidate different theta sketch representations into SketchHolder

### PR DESCRIPTION
**Pull request by himanshug**
Currently thetaSketch module has 3 different type of sketch representations that are passed around for performance reasons. This patch consolidates all the different sketch representations into single SketchHolder which handles all the messiness of different representations and remaining module only worries about SketchHolder.

Aside from refactoring, one major advantage of SketchHolder is to be able to cache the finalized sketch and estimates which improves performance of topN queries because they would add things to priority queue and estimates are computed repeatedly via comparator to place them in correct location on the priority queue. Its hard to quantify this one though.

This patch is fully backward and forward compatible.

### CODE REVIEW

**Comment by fjy** [comment]
👍

File: extensions-core/datasketches/src/main/java/io/druid/query/aggregation/datasketches/theta/SketchObjectStrategy.java [hunk 1]
```diff
@@ -35,46 +32,22 @@
 {
 
   private static final byte[] EMPTY_BYTES = new byte[]{};
-  private static final Sketch EMPTY_SKETCH = Sketches.updateSketchBuilder().build().compact(true, null);
+  private static final SketchHolder EMPTY_SKETCH = SketchHolder.of(
```
Comment (cheddar) [inline extensions-core/datasketches/src/main/java/io/druid/query/aggregation/datasketches/theta/SketchObjectStrategy.java]: I think this is a duplicate of the `public static` in `SketchHolder`

File: extensions-core/datasketches/src/main/java/io/druid/query/aggregation/datasketches/theta/SketchHolder.java [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+import io.druid.java.util.common.ISE;
+import org.apache.commons.codec.binary.Base64;
+
+import java.util.Comparator;
+
+/**
+ */
+public class SketchHolder
+{
+  public static final SketchHolder EMPTY_SKETCH = SketchHolder.of(
```
Comment (cheddar) [inline extensions-core/datasketches/src/main/java/io/druid/query/aggregation/datasketches/theta/SketchHolder.java]: This is a nit, but just the name `EMPTY` would make more sense to me.

File: extensions-core/datasketches/src/main/java/io/druid/query/aggregation/datasketches/theta/SketchObjectStrategy.java (hunk 1 above)
Comment (himanshug) [inline extensions-core/datasketches/src/main/java/io/druid/query/aggregation/datasketches/theta/SketchObjectStrategy.java]: removed

File: extensions-core/datasketches/src/main/java/io/druid/query/aggregation/datasketches/theta/SketchHolder.java (hunk 2 above)
Comment (himanshug) [inline extensions-core/datasketches/src/main/java/io/druid/query/aggregation/datasketches/theta/SketchHolder.java]: changed to EMPTY

**Pull request merged**